### PR TITLE
Add paymentMethodErrors, payerErrors, to PaymentDetailsUpdate

### LINKED ISSUE #766: PaymentDetailsUpdate needs to include payerErrors
There is bug in the current spec, whereby merchant can't signal payer-related fixes via `updateWith()`:

```JS
// user tries to fix email
response.onpayerdetailschange = ev => {
   if (!isValidPayerDetails(response)) {
       const payerErrors = getPayerErrors(response);
       // Oh wooops, `payerErrors` doesn't exist on `PaymentDetailsUpdate`
       ev.updateWith({ payerErrors });
   }
}
```

Same applies to `paymentMethod` errors.

### PR DESCRIPTION
**Pull request by marcoscaceres**
* closes #766 

The following tasks have been completed:

 * [x] Confirmed there are no ReSpec errors/warnings.
 * [ ] Modified Web platform tests (link)
 * [ ] Modified MDN Docs (link)

Implementation commitment:

 * [x] [Safari](https://bugs.webkit.org/show_bug.cgi?id=189938)
 * [ ] [Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=884433)
 * [x] [Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1435161) 
 * [ ] Edge (public signal)

Optional, Impact on Payment Handler spec?

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/pull/768.html" title="Last updated on Nov 5, 2018, 2:47 AM GMT (da086dc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/768/9f1244a...da086dc.html" title="Last updated on Nov 5, 2018, 2:47 AM GMT (da086dc)">Diff</a>

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@romandev, I saw you were helping implementing `.retry()` on the Chrome side. Is that correct?

File: index.html [hunk 1]
```diff
@@ -4725,14 +4745,32 @@ <h2>
...
                       specifically for each erroneous field of the shipping
                       address. This is done by matching each present member of
                       the <a>AddressErrors</a> to a corresponding input field
                       in the shown user interface.
                     </p>
+                    <p data-link-for="PaymentDetailsUpdate">
+                      Similarly, if <var>details</var>["<a>payerErrors</a>"]
+                      member is present and
+                      <var>request</var>.<a>[[\options]]</a>'s <a data-lt=
+                      "PaymentOptions.requestShipping">requestPayerName</a>,
```
Comment (romandev) [inline index.html]: nit: `<a data-lt="PaymentOptions.requestPayerName">`

File: index.html [hunk 2]
```diff
@@ -4725,14 +4745,32 @@ <h2>
...
+                      <a data-lt=
+                      "PaymentOptions.requestShipping">requestPayerEmail</a>,
```
Comment (romandev) [inline index.html]: nit: `<a data-lt="PaymentOptions.requestPayerEmail">`

File: index.html [hunk 3]
```diff
@@ -4725,14 +4745,32 @@ <h2>
...
+                      or <a data-lt=
+                      "PaymentOptions.requestShipping">requestPayerPhone</a> is
```
Comment (romandev) [inline index.html]: nit: `<a data-lt="PaymentOptions.requestPayerPhone">`

**Review by romandev — COMMENTED** [review]
Yep, I'm implementing `retry()` feature in Chromium under @rsolomakhin's high-quality review.

**Comment by marcoscaceres** [comment]
@romandev wrote:
> Yep, I'm implementing retry() feature in Chromium under @rsolomakhin's high-quality review.

Excellent to hear! What do you think of the proposed fix here? Make sense from an implementation perspective?

**Review by romandev — APPROVED** [review]
Sorry for delayed reply.
This seems reasonable to me for consistency.

One question is whether the some error values can be updated via ```updateWith()``` on an unrelated event.
For example, ```requestShipping``` and ```requestPayerName``` are all true and merchant calls ```updateWith()``` on payer detail change event as follows.
```js
  response.addEventListener('payerdetailchange', e => {
    e.updateWith({
      shippingAddressErrors: { city: 'CITY ERROR' }
    });
  });
```

In this case, should we allow ```shippingAddressErrors``` to be updated on payer detail change event?
FYI, in Chromium implementation, if the ```shippingAddressErrors``` is present, shipping address editor will perform after ```updateWith()``` calls.

Thank you.

**Comment by marcoscaceres** [comment]
> One question is whether the some error values can be updated via updateWith() on an unrelated event. For example, requestShipping and requestPayerName are all true and merchant calls updateWith() on payer detail change event as follows.

I think it's fine, tbh. 

> In this case, should we allow shippingAddressErrors to be updated on payer detail change event?

I think we should allow it - just to reduce our (implementation) complexity. It's not different from a developer randomly invalidating their `form` elements.

**Comment by marcoscaceres** [comment]
This is implemented in Firefox as part of https://bugzilla.mozilla.org/show_bug.cgi?id=1435161

**Comment by marcoscaceres** [comment]
@romandev, can I get implementation commitment from the Blink side?

**Comment by romandev** [comment]
@marcoscaceres Sure. I'm working on it.

**Comment by marcoscaceres** [comment]
@romandev awesome to hear. When you can, could you drop add a tracking bug? 🙏

**Comment by romandev** [comment]
I also filed this in Chromium :)
https://bugs.chromium.org/p/chromium/issues/detail?id=884433

**Comment by marcoscaceres** [comment]
We've implemented this in Firefox. Landing this with @romandev's r+.

**Comment by aestes** [comment]
We've implemented this in WebKit: https://trac.webkit.org/r236552

**Pull request merged**